submissions/the-darkesttt/frogger-task/app.js

```diff
@@ -0,0 +1,323 @@
+        this.menu.classList.remove('menu-animation');
+
+        this.menuDescriprion.innerText = 'You died! Your score:';
+
+        this.menuScore = document.createElement('span');
+        this.menuScore.innerText = points;
+        this.menuScore.classList.add('menu-score');
+
+        this.menu.appendChild(this.menuScore);
+
+        this.menuClose = document.createElement('button');
+        this.menuClose.innerText = 'Close menu and continue';
+        this.menuClose.classList.add('menu-close');
+        this.menu.appendChild(this.menuClose);
+
+        this.menuClose.addEventListener('click', (event) => {
+            this.bg.classList.add('menu-animation');
+            this.menu.classList.add('menu-animation');
+
+            this.menuDescriprion.innerText = '';
+            this.menuScore.parentElement.removeChild(this.menuScore);
+            this.menuClose.parentElement.removeChild(this.menuClose);
+        });
+    }
+}
+
+
+const startMenu = new Menu();
+startMenu.start();
+
+let points = 0;
+const score = document.createElement('p');
+score.classList.add('score');
+document.body.appendChild(score);
+
+function updateScore() {
+    score.innerHTML = 'Score: ' + points;
+}
+
+updateScore();
+
+function animateScoreGreen() {
+    score.classList.add('green');
+    setTimeout(function() {
+        score.classList.remove('green');
+    }, 1000)
+}
+
+function animateScoreBlue() {
+    score.classList.add('blue');
+    setTimeout(function() {
+        score.classList.remove('blue');
+    }, 1000)
+}
+
+let health = 3;
+
+const healthBar = document.createElement('div');
+document.body.appendChild(healthBar);
+healthBar.classList.add("health");
+
+function generateHealthBar() {
+
+    if (health <= 0) {
+        death();
+    }
+
+    if (healthBar.children.length > 0){
+        let imgArr = Array(...healthBar.children);
+        imgArr.forEach(item => {
+            item.remove();
+        });
+    }
+
+    for (let i = 1; i <= health; i++) {
+        let img = document.createElement('img');
+        img.setAttribute('src', 'images/Heart.png')
+        healthBar.appendChild(img);
+    }
+}
+
+generateHealthBar();
+
+function death() {
+    startMenu.death(points);
+
+    health = 3;
+    points = 0;
+}
+
+const blockSize = {
+    width: 101,
+    height: 85
+};
+
+const canvasSize = {
+    width: 505,
+    height: 606
+};
+
+const startPosition = {
+    x: blockSize.width * 2,
+    y: blockSize.width * 4
+}
+
+class Entity {
+    x;
+    y;
+    scrite;
+
+    constructor(x, y, sprite) {
+        this.x = x;
+        this.y = y;
+        this.sprite = sprite;
+    }
+
+    render() {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    }
+}
+
+class Enemy extends Entity {
+    speed = 100 + Math.floor(Math.random() * 222);
+
+    constructor(x, y, sprite) {
+        super(x, y, sprite);
+    }
+
+    handleInput(){}
+
+    update(dt) {
+
+        if (this.sprite == "images/enemy-bug.png") {
```

Comment:
It feels wrong to make this condition based on the sprite type. It would be better to add a property to enemies, like `facingDirection`, that would just be left/right.